File: leetcode3075.py
class Solution(object):
    def maximumHappinessSum(self, happiness, k):
        happiness.sort(reverse=True)
        i=0
        sum=0

        while k>0:
            happiness[i]= max(happiness[i]-i,0)
            sum+=happiness[i]
            i+=1
            k-=1
        
        return sum
        

File: leetcode3075.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness);
        long sum = 0;
        int index=0;
        for(int i=happiness.length-1;i>=happiness.length-k;i--){
            happiness[i]=Math.max(happiness[i]-index++,0);
            sum+=happiness[i];
        }
        
        return sum;
    }
}
